snuba/migrations/snuba_migrations/generic_metrics/0002_sets_raw_table.py

```diff
@@ -0,0 +1,89 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import (
+    Array,
+    Column,
+    DateTime,
+    Float,
+    Nested,
+    String,
+    UInt,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+    local_table_name = "generic_metric_sets_raw_local"
+    dist_table_name = "generic_metric_sets_raw_dist"
+    columns: Sequence[Column[Modifiers]] = [
+        Column("use_case_id", String(Modifiers(low_cardinality=True))),
+        Column("org_id", UInt(64)),
+        Column("project_id", UInt(64)),
+        Column("metric_id", UInt(64)),
+        Column("timestamp", DateTime()),
+        Column("retention_days", UInt(16)),
+        Column(
+            "tags",
+            Nested(
+                [
+                    ("key", UInt(64)),
+                    ("indexed_value", UInt(64)),
+                    ("raw_value", String()),
+                ]
+            ),
+        ),
+        Column("set_values", Array(UInt(64))),
+        Column("count_value", Float(64)),
+        Column("distribution_values", Array(Float(64))),
+        Column("metric_type", String(Modifiers(low_cardinality=True))),
+        Column("materialization_version", UInt(8)),
+        Column("timeseries_id", UInt(32)),
+        Column("partition", UInt(16)),
+        Column("offset", UInt(64)),
+    ]
+
+    def forwards_local(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=StorageSetKey.GENERIC_METRICS_SETS,
+                table_name=self.local_table_name,
+                engine=table_engines.MergeTree(
+                    storage_set=StorageSetKey.GENERIC_METRICS_SETS,
+                    order_by="(use_case_id, org_id, project_id, metric_id, timestamp)",
+                    partition_by="(toStartOfInterval(timestamp, toIntervalDay(3)))",
+                    ttl="timestamp + toIntervalDay(7)",
```

Comment:
Was there a specific reason you chose to have partitions of 3 days and retention of 7?

Reply:
I think the reasoning for the 3 days partitioning comes from https://github.com/getsentry/snuba/pull/2524